snapshot/src/main/java/io/zeebe/snapshots/broker/impl/FileBasedSnapshotStore.java

```diff
@@ -71,7 +71,14 @@ private FileBasedSnapshot loadLatestSnapshot(final Path snapshotDirectory) {
     FileBasedSnapshot latestPersistedSnapshot = null;
     try (final var stream = Files.newDirectoryStream(snapshotDirectory)) {
       for (final var path : stream) {
-        latestPersistedSnapshot = collectSnapshot(path);
+        final var snapshot = collectSnapshot(path);
+        if (latestPersistedSnapshot == null) {
+          latestPersistedSnapshot = snapshot;
+        } else if (snapshot != null
```

Comment:
Couldn't we delete the snapshot here if it's less recent than the `latestPersistedSnapshot`?

Reply:
Yes. I fixed it now. I now refactored it to make it explicit that we are deleting all older snapshot.